src/js/components/epics/detail.tsx

```diff
@@ -372,19 +374,47 @@ const EpicDetail = (props: RouteComponentProps) => {
   const onRenderHeaderActions = () => (
     <PageHeaderControl>
       {project.has_push_permission && (
-        <PageOptions
-          modelType={OBJECT_TYPES.EPIC}
-          handleOptionSelect={handlePageOptionSelect}
-        />
+        <div className="slds-is-relative inline-container vertical-right">
+          <PageOptions
+            modelType={OBJECT_TYPES.EPIC}
+            handleOptionSelect={handlePageOptionSelect}
+          />
+          <TourPopover
+            align="left"
+            heading={i18n.t('Edit & delete the Epic')}
+            body={
+              <Trans i18nKey="tourEditEpic">
+                Here you can change the name and description of this Epic. You
+                can also delete the Epic. Deleting an Epic deletes all the Tasks
+                and Orgs in the Epic as well.
+              </Trans>
+            }
+          />
+        </div>
       )}
       {branchLink && (
-        <ExternalLink
-          url={branchLink}
-          showButtonIcon
-          className="slds-button slds-button_outline-brand"
-        >
-          {branchLinkText}
-        </ExternalLink>
+        <div className="slds-is-relative inline-container vertical-right">
+          <ExternalLink
+            url={branchLink}
+            showButtonIcon
+            className="slds-button slds-button_outline-brand"
+          >
+            {branchLinkText}
+          </ExternalLink>
+          <TourPopover
+            align="bottom right"
+            heading={i18n.t('View GitHub branch for Epic')}
+            body={
+              <Trans i18nKey="tourViewEpicBranch">
+                Select this button to leave Metecho and access the Epic’s branch
+                on GitHub. A “branch” in Git is a way to create a new feature or
+                make a modification to existing software but not affect the main
+                “trunk” of the Project. A branch is created in GitHub when a new
+                Epic or Task is created in Metecho.
+              </Trans>
+            }
+          />
```

Comment:
Both here (epic detail) and on task detail, we want this popover to change depending whether the button is "View Branch", "View Changes", or "View Pull Request". You can see the logic for this in `helpers.ts`, `getBranchLink` fn. Maybe we want a similar fn to get the right popover? Or we could repurpose `getBranchLink` to also return the popover?

Reply:
A little stuck here. Having a hard time figuring out the syntax to test whether the object is an epic or a task. Is what I'm doing here so far even close to being on the right track?

Reply:
@dvdherron Yeah, I agree this is a bit tricky, but what you have is definitely on the right track. There would be a number of different ways to approach this, but I'd say the clearest might be adding a second argument that `getBranchLink` accepts: `type: 'epic' | 'task'`. Then I think in the fn body, you could do e.g. `taskPopoverHeading = type === OBJECT_TYPES.EPIC ? i18n.t('View GitHub Branch for Epic') : i18n.t('View GitHub Branch for Task');` -- or something along those lines?

Happy to pair with you if that helps. I'd also say if you get the logic working but are struggling to make TypeScript warnings happy, don't spend much time on that. Just commit and I can look closer at the TS.

Reply:
Thanks. My meeting went longer than planned. Digging into this now.

Reply:
was able to get this to work. will give it another look in the AM to see if there's a better way @jgerigmeyer